### ISSUE TITLE
Unable to install extension pack

### ISSUE THREAD

**Issue by KishoreIthadi** [Microsoft/vscode#50385]
An extension pack (https://marketplace.visualstudio.com/items?itemName=salbert.awesome-dotnetcore-pack), which contains DotNet-Core-Essentials. Every time when I change the extension pack and deploy the new version, "dotnet-core-essentials" tries to update version 0.0.4 and fails, which implies that the extension pack also fails. The workaround is to uninstall DotNet-Core-Essentials and install the extension pack. Then the most recent version of dotnet-core-essentials is installed, but next time the extension pack is updated (without changing DotNet-Core-Essentials), it tries to update DotNet-Core-Essentials again and fails.... I tested this behaviour on two different pcs.

![12](https://user-images.githubusercontent.com/19744167/40473323-703a8320-5f59-11e8-8937-203236c6ea03.gif)

On investigating further found below error

![image](https://user-images.githubusercontent.com/19744167/40473162-09699280-5f59-11e8-8a0d-56d9c76262b2.png)

**Comment by weinand** [comment]
@sandy081 is this a VS Code or extension issue?

**Comment by KishoreIthadi** [comment]
yes,
if you closely observe the gif from or the previous comment, 
"Awesome DotNetCore Pack" extension pack is not getting installed if "DotNet-Core-Essentials" extension is already installed.

**Comment by sandy081** [comment]
Looks like an issue with VS Code. But have to investigate for why.

**Comment by sandy081** [comment]
@KishoreIthadi I do not see `Dotnet Core Essentials` extension is listed as dependency in the pack. If it is not listed, then it will not be installed. Was it removed?

**Comment by KishoreIthadi** [comment]
@sandy081 yes it is has been removed since that extension is unable to update 

Here is that commit
https://github.com/s-albert/awesome-dotnetcore-pack/commit/f206cf88c5bd959d4e14d06e4b68006622477627

**Comment by sandy081** [comment]
Can you please let me know if you are able to reproduce it yet?

**Comment by KishoreIthadi** [comment]
Yes @sandy081. I have downloaded the repo and created a vsix file. The issue still exists. Installing the attached vsix file is giving this error

[awesome-dotnetcore-pack-1.5.0.zip](https://github.com/Microsoft/vscode/files/2062012/awesome-dotnetcore-pack-1.5.0.zip)

![error](https://user-images.githubusercontent.com/19744167/40832061-37c08562-65a8-11e8-8d34-58da51a5f469.png)

**Comment by sandy081** [comment]
Thanks for the vsix. I will try to repro and investigate

**Comment by sandy081** [comment]
@KishoreIthadi I just tried it and everything seems ok for me. 

- Installed `dotnet-core-essentials` extension 
- Installed the attached vsix

All installed. Which version of VS Code are you trying?

**Comment by KishoreIthadi** [comment]
I have published the extension as "dcetest", https://marketplace.visualstudio.com/items?itemName=DCETest.dcetest

Could you please try with this?

**Comment by sandy081** [comment]
It worked.

**Comment by KishoreIthadi** [comment]
it is not working for me. I am using windows OS, are you using the same?

**Comment by sandy081** [comment]
I am on Mac though. Can you do following...

- Set the log level to `trace`. Use `Set Log Level` command
- Reproduce the bug again
- Get the Shared process logs. Use `Show logs` command

**Comment by KishoreIthadi** [comment]
@sandy081 here are the logs

[log4.txt](https://github.com/Microsoft/vscode/files/2075011/log4.txt)
[log1.txt](https://github.com/Microsoft/vscode/files/2075012/log1.txt)
[log2.txt](https://github.com/Microsoft/vscode/files/2075014/log2.txt)
[log3.txt](https://github.com/Microsoft/vscode/files/2075015/log3.txt)

**Comment by KishoreIthadi** [comment]
@sandy081 Could you please take a look at this and resolve ASAP.

**Comment by sandy081** [comment]
@KishoreIthadi I see following error 

`Failed renaming C:\Users\kishore.ithadi\.vscode\extensions\.KishoreIthadi.dotnet-core-essentials-0.0.5 to C:\Users\kishore.ithadi\.vscode\extensions\KishoreIthadi.dotnet-core-essentials-0.0.5 with 'EPERM' error.`

Can you please check in your system why renaming above folder was not getting permitted?

Do you have any software like Defender that is holding on to this folder?

**Comment by KishoreIthadi** [comment]
@sandy081 I have tested this in the system without any defender/antivirus, but it is behaving the same way

**Comment by sandy081** [comment]
Sorry for asking so many questions. I have to as I am not able to reproduce and have to get as much information as I can to understand and repro it on my side.

Do you have `KishoreIthadi.dotnet-core-essentials-0.0.5` folder exists already in extensions folder before installing?

**Comment by KishoreIthadi** [comment]
Not a problem @sandy081. 

Yes, the folder exists. Just to recall the issue the "DCE" extension is already installed so "KishoreIthadi.dotnet-core-essentials-0.0.5" folder will be present. Later user is trying to install "Awesome" extension and unable to install it.

**Comment by sandy081** [comment]
Ok.. looks like the extension is getting installed again which is why it is failing.

**Comment by sandy081** [comment]
Can you please check if the version of the extension in the package.json in `KishoreIthadi.dotnet-core-essentials-0.0.5` before installing the extension pack ? And is it the same version that the extension pack is installing?

**Comment by KishoreIthadi** [comment]
yes it is the same version '0.0.5'

**Comment by sandy081** [comment]
Looks related #52946..

Pushed a fix. Please try tomorrows insiders if it is fixed or not. Thanks.
